src/GafferScene/InteractiveRender.cpp

```diff
@@ -57,6 +61,71 @@ using namespace GafferScene;
 
 IE_CORE_DEFINERUNTIMETYPED( InteractiveRender );
 
+//////////////////////////////////////////////////////////////////////////
+// SceneGraph implementation
+//
+// This is a node in a scene hierarchy, which gets built when the
+// interactive render starts up. Each node stores the attribute hash of
+// the input scene at its corresponding location in the hierarchy, so
+// when the incoming scene updates, we are able to determine the locations
+// at which the attributes have changed since the last update, reevaluate
+// those attributes and send updates to the renderer.
+//
+// \todo: This is very similar to the SceneGraph mechanism in
+// GafferSceneUI::SceneGadget. At some point it would be good to refactor
+// this and use the same class for both of them. See the comments in
+// src/GafferSceneUI/SceneGadget.cpp for details.
+//
+//////////////////////////////////////////////////////////////////////////
+
+class InteractiveRender::SceneGraph
+{
+
+	public :
+
+		SceneGraph() : m_parent(0)
```

Comment:
I know some of the older code in Gaffer is using `0` rather than `NULL`, but all new code should use `NULL`. Then we can move to using the nice new `nullptr` when we get to use C++11 compilers.